src/js/messaging/components/forms/NewMessageForm.jsx

```diff
@@ -14,20 +14,42 @@ import MessageWriteGroup from '../compose/MessageWriteGroup';
 export class NewMessageForm extends React.Component {
   render() {
     const message = this.props.message;
+    let recipientsField;
 
     // Tests the subject group for errors
     const subjectError = validations.isValidSubjectLine(message.category, message.subject);
 
+    if (this.props.recipients) {
+      if (this.props.recipients.length) {
+        recipientsField = (
+          <MessageRecipient
+              errorMessage={validations.isValidRecipient(message.recipient) ? '' : composeMessage.errors.recipient}
+              cssClass="msg-recipient msg-field"
+              onValueChange={this.props.onRecipientChange}
+              options={this.props.recipients}
+              recipient={message.recipient}/>
+        );
+      } else {
+        // The list of recipients somehow came back empty.
+        recipientsField = <p>No recipients could be found.</p>;
```

Comment:
Is this case meaningfully different from `recipients` being falsy? The comment suggests it's not clear how you would get here, so is there a good reason to hide the retry link?

Also, may just be me, but I find the nested if else blocks here hard to read. This would be clearer to me:

```
  if (this.props.recipients && this.props.recipients.length) {
      recipientsField = (
        <MessageRecipient
            errorMessage={validations.isValidRecipient(message.recipient) ? '' : composeMessage.errors.recipient}
            cssClass="msg-recipient msg-field"
            onValueChange={this.props.onRecipientChange}
            options={this.props.recipients}
            recipient={message.recipient}/>
      );
    } else {
      // only show the retry option if recipients is falsy, which means the request failed
      recipientsField = (
        <p>
          No recipients could be found.&nbsp;
          {!!this.props.recipients && <a onClick={this.props.onFetchRecipients}>
            Click here to retry loading recipients.
          </a>}
        </p>
      );
    }
```